### PR TITLE
No more singleton

### PR DESCRIPTION
**Pull request by wbowling**
We required multiple clients with different access tokens which wasn't possible before

### CODE REVIEW

File: shutterstock-ruby.gemspec [hunk 1]
```diff
@@ -4,7 +4,7 @@ require 'date'
 
 Gem::Specification.new do |s|
   s.name        = 'shutterstock-ruby'
-  s.version     = '0.2.0'
+  s.version     = '1.0.0'
```
Comment (jkburges) [inline shutterstock-ruby.gemspec]: Is this PR a breaking change?  Is it worth talking to upstream maintainer if so (or make it backward compatible, if it's not too hard)?

File: lib/shutterstock-ruby/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module ShutterstockRuby
+  class Configuration
+    attr_accessor :access_token
```
Comment (jkburges) [inline lib/shutterstock-ruby/configuration.rb:3]: Is `attr_reader` enough for these 3?

File: shutterstock-ruby.gemspec (hunk 1 above)
Comment (wbowling) [reply shutterstock-ruby.gemspec]: Yeah everything was static before so it's a breaking change. I'll have a look to see if I can make it backwards compatible, but not sure if it's possible

File: lib/shutterstock-ruby/configuration.rb (hunk 2 above)
Comment (wbowling) [reply lib/shutterstock-ruby/configuration.rb:3]: Yeah could be, I was thinking you might want to change them after creating but on second thoughts that probably wont need to happen

File: shutterstock-ruby.gemspec (hunk 1 above)
Comment (jkburges) [reply shutterstock-ruby.gemspec]: Might be able to keep the static stuff, and initialise a `Configuration` with those values if present?  Or just talk to the maintainer and see if s(he) thinks it's ok to have a breaking change.

File: lib/shutterstock-ruby/configuration.rb (hunk 2 above)
Comment (wbowling) [reply lib/shutterstock-ruby/configuration.rb:3]: `attr_accessor` is required for the backwards compatibility

**Comment by wbowling** [comment]
Ok wasn't too hard, now should be non breaking and allow for multiple instances if required

File: shutterstock-ruby.gemspec [hunk 3]
```diff
@@ -4,7 +4,7 @@ require 'date'
 
 Gem::Specification.new do |s|
   s.name        = 'shutterstock-ruby'
-  s.version     = '0.2.0'
+  s.version     = '0.4.0'
```
Comment (jkburges) [inline shutterstock-ruby.gemspec:7]: I assume there's a `0.3.x` upstream?

Comment (wbowling) [reply shutterstock-ruby.gemspec:7]: yeah, our master has diverged. If this gets merged then we can reset our master to be the same as theirs. I didn't want to just change it as I think the we app is currently pointing to here and might be looking for a certain commit

**Comment by jkburges** [comment]
☘️

**Comment by wbowling** [comment]
Upstream PR merged https://github.com/TailorBrands/shutterstock-ruby/pull/6

**Pull request closed without merging**